lib/registries/addon/overview/controller.ts

```diff
@@ -2,22 +2,65 @@ import { action, computed } from '@ember-decorators/object';
 import { alias, not } from '@ember-decorators/object/computed';
 import { service } from '@ember-decorators/service';
 import Controller from '@ember/controller';
+import config from 'ember-get-config';
+import I18N from 'ember-i18n/services/i18n';
 import Media from 'ember-responsive';
 
 import Registration from 'ember-osf-web/models/registration';
 import { GuidRouteModel } from 'ember-osf-web/resolve-guid/guid-route';
+import CurrentUser from 'ember-osf-web/services/current-user';
+import pathJoin from 'ember-osf-web/utils/path-join';
+
+enum RegistrationState {
+    EMBARGOED = 'embargoed',
+    PUBLIC = 'public',
+    WITHDRAWN = 'withdrawn',
+    PENDING_REGISTRATION = 'pendingRegistrationApproval',
+    PENDING_WITHDRAWAL = 'pendingWithdrawal',
+    PENDING_EMBARGO = 'pendingEmbargoApproval',
+    PENDING_EMBARGO_TERMINATION = 'pendingEmbargoTerminationApproval',
+}
+
+const { OSF: { url: baseURL } } = config;
 
 export default class Overview extends Controller {
     @service media!: Media;
-    @not('media.isDesktop') showMobileNav!: boolean;
+    @service currentUser!: CurrentUser;
+    @service i18n!: I18N;
 
     model!: GuidRouteModel<Registration>;
 
     sidenavGutterClosed = true;
     metadataGutterClosed = true;
 
-    @alias('model.taskInstance.value') registration?: Registration;
+    @not('media.isDesktop') showMobileView!: boolean;
     @not('registration') loading!: boolean;
+    @alias('model.taskInstance.value') registration!: Registration;
+    @alias('registration.userHasAdminPermission') isAdmin!: boolean;
+
+    @computed('registration.id')
+    get registrationURL() {
+        return this.registration && pathJoin(baseURL, `${this.registration.id}`);
+    }
+
+    /* eslint-disable max-len */
+    @computed('registration.{withdrawn,embargoed,public,pendingRegistrationApproval,pendingEmbargoApproval,pendingEmbargoTerminationApproval,pendingWithdrawal}')
+    get currentState() {
```

Comment:
Could you move this CP to the `Registration` model, and put the `RegistrationState` enum there too? There are two things that would get you:
1. Be able to pass only `registration` to most of the registration-specific components, instead of `registration`, `isAdmin`, `isRoot`, and `currentState`
2. Import `RegistrationState` from a central place, for e.g. the various `StateToFooMap`s you've added